src/gencost/waterfall.py

```diff
@@ -2134,125 +2218,270 @@ def validate_merge_all_results(merge_all_df):
+                    # "cum_starts": Column(int, Check.ge(0)),
+                    "pollution_control_costs_per_kw": Column(float, Check.ge(0.0)),
+                    "real_pollution_control_costs_per_kw": Column(float, Check.ge(0.0)),
+                    "wage_scale": Column(float),
+                }
+                | {
+                    "age_of_observation_secular_adj": Column(float),
+                    "age_of_observation": Column(float, Check.in_range(0.0, 2e3)),
+                    "age_relative_to_prime_avg": Column(float),
+                }
+                | {
+                    f"{k}_fraction": Column(float, Check.in_range(0.0, 1.0))
+                    for k in fuels
+                }
+                | {"minor_fuels_fraction": Column(float, Check.in_range(0.0, 1.0))}
+                | {k: Column(float, Check.in_range(0.0, 1.0)) for k in techs}
+                # not used in regression
+                | {
+                    "age_in_report_year": Column(float),
+                    "age_in_current_year": Column(float, Check.in_range(0.0, 2e3)),
+                    # "parasitic_load_pct": Column(float),
+                    # "camd_capacity_mw": Column(float, Check.in_range(0.0, 1e4)),
+                    "gross_generation_mwh": Column(float, Check.ge(0.0)),
+                    # "gross_hr": Column(float, Check.ge(0.0), nullable=True),
+                    # "heat_in_mmbtu": Column(float, Check.ge(0.0)),
+                    "net_generation_mwh": Column(float),
+                    # "net_cf": Column(float, nullable=True),
+                    # "arc": Column(float, nullable=True),
+                    "inflator_to_2021": Column(float),
+                    "fuel_starts": Column(int, Check.ge(0)),
+                    # "opex": Column(float, Check.ge(0.0), nullable=True),
+                    # "real_capex": Column(float, Check.ge(0.0), nullable=True),
+                    # "opex_per_kw": Column(float, Check.ge(0.0), nullable=True),
+                    # "capex_per_kw": Column(float, Check.ge(0.0), nullable=True),
+                }
+                | {
+                    f"{k}_mmbtu": Column(float, Check.ge(0.0), nullable=True)
+                    for k in fuels
+                }
+                | {f"{k}_net_mwh": Column(float, nullable=True) for k in fuels}
+                | {
+                    f"{k}_gross_mwh": Column(float, Check.ge(0.0), nullable=True)
+                    for k in fuels
+                }
+                # {f"{k}_gross_cf": Column(float, Check.ge(0.0)) for k in fuels}
+            )
 
-        def x_gen_allocation(df_, kind):
-            return pd.Series(
-                np.isclose(
-                    df_.filter(like=f"_{kind}_mwh").sum(axis=1),
-                    df_[f"{kind}_generation_mwh"],
-                    rtol=2e-2,
-                ),
-                index=df_.index,
+            def gross_ge_net(df_):
```

Comment:
Let's move all of these functions within `core_validator` to before the `if` so they only need to be defined once. Same with the `fuels`, `techs`, and `core_columns`.